tests/Avalonia.Controls.UnitTests/GridTests.cs

```diff
@@ -155,6 +157,68 @@ public void Layout_StarRowColumnWithMaxLength_BoundsCorrect()
             GridAssert.ChildrenWidth(columnGrid, 200, 300, 300);
         }
 
+        [Fact]
+        public void Layout_StarColumn_Should_Pass_Calculated_Width_On_Measure_And_Arrange()
+        {
+            Square child;
+
+            var target = new Grid
+            {
+                ColumnDefinitions = new ColumnDefinitions("*,*"),
+                Children =
+                {
+                    (child = new Square
+                    {
+                        [Grid.ColumnProperty] = 0,
+                        HorizontalAlignment = HorizontalAlignment.Center,
+                        VerticalAlignment = VerticalAlignment.Center
+                    })
+                }
+            };
+
+            target.Measure(new Size(200, 200));
+            target.Arrange(new Rect(target.DesiredSize));
+
+            Assert.Equal(target.ColumnDefinitions[0].ActualWidth, 100);
+            Assert.Equal(target.ColumnDefinitions[1].ActualWidth, 100);
+
+            Assert.Equal(new Size(100, 100), child.DesiredSize);
+            Assert.Equal(new Rect(0, 50, 100, 100), child.Bounds);
+        }
+
+        [Fact]
+        public void Layout_LeftAlingment_And_StarColumn_Shouldnt_Stretch()
+        {
+            Square child;
+
+            var target = new Grid
+            {
+                ColumnDefinitions = new ColumnDefinitions("*,100"),
+                HorizontalAlignment = HorizontalAlignment.Left,
+                Children =
+                {
+                    (child = new Square
+                    {
+                        [Grid.ColumnProperty] = 0,
+                        HorizontalAlignment = HorizontalAlignment.Center,
+                        VerticalAlignment = VerticalAlignment.Center
+                    })
+                }
+            };
+
+            target.Measure(new Size(300, 100));
+            target.Arrange(new Rect(target.DesiredSize));
+
+            Assert.Equal(target.ColumnDefinitions[0].ActualWidth, 100);
```

Comment:
The expected argument should be on the left and the actual value should be on the right. So as the below.